Libraries/libbom/Sources/bom_tree.c

```diff
@@ -228,15 +228,51 @@ bom_tree_add(struct bom_tree_context *tree_context, const void *key, size_t key_
         paths = (struct bom_tree_entry *)bom_index_get(tree_context->context, paths_index, &paths_length);
     }
 
-    size_t entry_index = ntohs(paths->count);
+    size_t last_index = ntohs(paths->count);
+    size_t entry_index = 0;
+    size_t start_range = 0;
+    size_t end_range = last_index;
 
-    /* Set the indexes for the inserted entry. */
+    while (start_range < end_range) {
```

Comment:
Would be nice to have a comment here what this block is actually doing.

Reply:
Can do.  This is, if I remember correctly, supposed to find the position in the BOM tree that the new entry sorts to (since it needs to be sorted for fast lookup).